quantumblacklabs/kedro-viz
# [KED-2831] Fix focus mode error when switching pipelines

## Description

related ticket: https://jira.quantumblack.com/secure/RapidBoard.jspa?rapidView=525&projectKey=KED&view=detail&selectedIssue=KED-2831

This is to fix the error thrown when switching between pipelines while in focus mode. 

## Development notes

The bug is caused by the persistence of the focus mode state while switching between pipelines - the problem will be solved by resetting the value of the selected focus mode modular pipeline when switching between pipelines. 

I have added the dispatch of the relevant action, as well as the relevant test to ensure this action to reset the focus mode state gets dispatched everytime we switch between pipelines. 

## QA notes

select a modular pipeline to be under focus mode, and try switching between pipelines. The error should not happen again. 

## Checklist


## Legal notice


- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
